### ISSUE TITLE
Explore all possible solution to improve the training and loading of the saved model.

### ISSUE THREAD

**Issue by afmjoaa** [nerdslab-club/cl_model#27]
- [x] Category router-trained params aren't saved perfectly
- [x] Make Grad = False for the tokenizer model hidden layer
- [x] Don't backpropagate on the same graph multiple times, use a different loss function every time.
- Approach one: Combine all loss calculations in a loss function and backpropagate once
- Approach two: Use different separate loss functions, but figure out a way to combine them.
